Add Calendar Model

lihongguang00 opened the pull request:
Added model for calendar, along with addEvent feature.

Storage feature for events not implemented yet.

Commit:
Migrate docs to MarkBind

Let's migrate the docs site from Jekyll to MarkBind.

Primary author: @tlylt in se-edu/addressbook-level3/pull/156
Further tweaks: @damithc in se-edu/addressbook-level3/pull/206

lihongguang00 committed:
Add Event Model

Created model for event for calendar

lihongguang00 committed:
Add Event JavaDoc

Add JavaDoc for Event class

lihongguang00 committed:
Calendar WIP

Work in Progress

lihongguang00 committed:
DG User Story

Added User Stories

lihongguang00 committed:
DG Week7

Completed Requirements section of DG for UniMate

junhonglow committed:
Add skeletal PPP

lihongguang00 committed:
Add Basic CML Calendar

Added basic CML Calendar

Included features:
- Add event

lihongguang00 committed:
Add JavaDoc

Added JavaDocs to the following
-AddEventCommandParser
-AddressBookParser
-CalendarParser
-ComponentParser
-Calendar
-Event
-EventDecsription
-EventPeriod

junhonglow committed:
Update ReadME

junhonglow committed:
Resolve checkstyle error

lihongguang00 committed:
Fix Checkstyle

Fixed Checkstyle and added more javadocs

lihongguang00 committed:
Fix JUnit test bug

Added necessary methods to fix JUnit test bugs

lihongguang00 committed:
Add AddEventCommand JUnit Test

Added JUnit test for AddEventCommand

lihongguang00 committed:
Update dependancies for AddEventCommand JUnit Testing

Included updated dependencies for AddEventCommand JUnit Testing

lihongguang00 committed:
Fix AddEventCommandTest checkstyle

Fixed checkstyle for AddEventCommandTest

lihongguang00 committed:
Fix AddEventCommandTest checkstyle v2

Fixed AddEventCommandTest checkstyle

lihongguang00 committed:
Fix AddEventCommandTest checkstyle v3

Fixed AddEventCommandTest checkstyle

lihongguang00 committed:
Add ParserUtil JUnit testing

Added JUnit testing for:
-parseEventDescription
-parseEventPeriod

lihongguang00 committed:
Fix checkstyle for ParserUtilTest

Fixed checkstyle erros for ParserUtilTest

lihongguang00 committed:
Update AddEvent UserGuide

Updated user guide for AddEvent command to reflect new syntax

lihongguang00 committed:
Add JUnit testing for AddEventCommandParser

Added JUnit testing for AddEventCommandParser

lihongguang00 committed:
Fixed AddEventCommandParserTest checkstyle

Fixed checkstyle for AddEventCommandParserTest

lihongguang00 committed:
Fixed AddEventCommandParserTest checkstyle v2

Fixed checkstyle for AddEventCommandParserTest

lihongguang00 committed:
Add Calendar JUnit test

Added JUnit testing for Calendar class

lihongguang00 committed:
Restructure Parser

Deleted CalendarParser and ApplicationParser
Integerated all parsers into UniMateParser

lihongguang00 committed:
Add More JUnit Testing for Calendar

Added more test cases for testing Calendar class

lihongguang00 committed:
Fixed checkstyle for more Calendar JUnit testing

Fixed Checkstyle for more Calendar JUnit testing

lihongguang00 committed:
Restructure Calendar

Abstracted out:
-Daily events into a single SingleDayEventList class
-Collection of SingleDayEventList class into AllDaysEventListManager

Additionally, changed underlying data structure storing events in Calendar class into a single AllDaysEventListManager

lihongguang00 committed:
Fix Checkstyle

Fixed checkstyle for previous commit

lihongguang00 committed:
Fix even more Checkstyle

Fixed even more checkstyle for previous commit

lihongguang00 committed:
Fix addEvent command

Fix logic error causing addEvent command to not work

lihongguang00 committed:
FIx AddEventCommandParser JUnit testing

Fixed JUnit testing for AddEventCommandParser

lihongguang00 committed:
Fix AddEventCommandParserTest checkstyle

Fixed Checkstyle for AddEventCommandParserTest

lihongguang00 committed:
Added JUnit Test for EventPeriod class

Added 100% code coverage for EventPeriod class

lihongguang00 committed:
Fix Checkstyle for EventPeriodTest

Fixed Checkstyle for EventPeriodTest

lihongguang00 committed:
Fix more Checkstyle for EventPeriodTest

Fixed more checktyle for EventPeriodTest

lihongguang00 committed:
Fix more Checkstyle for EventPeriodTest

Fixed more checktyle for EventPeriodTest

lihongguang00 committed:
Fix more Checkstyle for EventPeriodTest

Fixed even more checkstyle for EventPeriodTest

lihongguang00 committed:
Add JUnit testing for EventDescription

Add full code coverage for EventDescription class

lihongguang00 committed:
Add JUnit testing for Event

Added JUnit testing for Event class

lihongguang00 committed:
Fix JUnit testing for Event

Fix JUnit testing for Event class

lihongguang00 committed:
Partial JUnit Testing for SingleDayEventList

lihongguang00 committed:
Partial JUnit Testing for SingleDayEventList

lihongguang00 committed:
Add JUnit Testing for SingleDayEventList class

Added full code coverage for SingleDayEventList class

Fallman2 commented on src/main/java/seedu/address/logic/LogicManager.java:
```diff
@@ -31,23 +31,23 @@ public class LogicManager implements Logic {
         this.model = model;
         this.storage = storage;
-        addressBookParser = new AddressBookParser();
+        this.uniMateParser = new UniMateParser();
```
The keyword "this' can be removed.

Fallman2 commented on src/main/java/seedu/address/logic/commands/AddEventCommand.java:
```diff
@@ -0,0 +1,80 @@
+
+    public static final String MESSAGE_SUCCESS = "New event added: %1$s";
+
+    public static final String MESSAGE_EVENT_CONFLICT = "This event is conflicting with another event";
```
Add a full-stop or ! after the sentence to keep it consistent with other messages.

lihongguang00 replied:
For `AddCommand` given in AB3, theres no full-stop and ! after the sentence too

Fallman2 commented on src/main/java/seedu/address/logic/parser/ParserUtil.java:
```diff
@@ -121,4 +124,38 @@ public static Set<Tag> parseTags(Collection<String> tags) throws ParseException
+        requireNonNull(description);
+        String trimmedDescription = description.trim();
+        if (trimmedDescription.isEmpty()) {
+            throw new ParseException(EventDescription.MESSAGE_CONSTRAINTS);
```
Consider putting this check into the EventDescription constructor instead as a validity check.

lihongguang00 replied:
Abstracted out the checking of whether the `trimmedDescription` is a valid description by adding a static method in `EventDescription` `isValid(String)` to check.

Will still keep the check to be consistent with `parseIndex`, `parseName`, `parsePhone`, `parseAddress` and `parseEmail` methods in the same `ParserUtil.java` file

Fallman2 commented on src/main/java/seedu/address/logic/parser/ParserUtil.java:
```diff
@@ -121,4 +124,38 @@ public static Set<Tag> parseTags(Collection<String> tags) throws ParseException
+        String trimmedStartDate = startDate.trim();
+        String trimmedEndDate = endDate.trim();
+        if (!EventPeriod.isValidPeriod(trimmedStartDate, trimmedEndDate)) {
+            throw new ParseException(EventPeriod.MESSAGE_CONSTRAINTS);
```
Same as above. Consider putting this check into EventPeriod constructor instead.

lihongguang00 replied:
Will still keep the check to be consistent with `parseIndex`, `parseName`, `parsePhone`, `parseAddress` and `parseEmail` methods in the same `ParserUtil.java` file

Fallman2 commented on src/main/java/seedu/address/logic/parser/ComponentParser.java:
Since there is only 1 class that extends Component Parser it might be better to put the methods of this class into UniMateParser instead.

Fallman2 commented on src/main/java/seedu/address/model/ModelManager.java:
```diff
@@ -111,6 +118,19 @@ public void setPerson(Person target, Person editedPerson) {
+    //=========== Calendar ===================================================================================
+    @Override
+   public boolean canAddEvent(Event event) {
+        return this.calendar.canAddEvent(event);
```
Can remove "this" keyword here.

Fallman2 commented on src/main/java/seedu/address/model/ModelManager.java:
```diff
@@ -111,6 +118,19 @@ public void setPerson(Person target, Person editedPerson) {
+    public void addEvent(Event event) {
+        requireAllNonNull(event);
+
+        this.calendar.addEvent(event);
```
Can remove "this" here as well.

Fallman2 commented on src/main/java/seedu/address/model/event/EventPeriod.java:
```diff
@@ -0,0 +1,142 @@
+     * @param other The EventPeriod to compare with.
+     * @return 1 if this EventPeriod is after the other, -1 if it's before, 0 if they are the same.
+     */
+    public int compareTo(EventPeriod other) {
```
Since there is a compareTo method, this class might want to implement Comparable<EventPeriod>

Fallman2 reviewed (commented):
Generally looks good aside from a few small formatting issues.

lihongguang00 committed:
Resolve Fallman2 review comments

Resolved all comments except for:
-ParserUtil lines 136-139
-ParserUtil lines 154-157

lihongguang00 committed:
Deleted Unnecessary Files and Resolved Merge Conflict

Deleted johndoe.md and resolved merge conflicts with origin master

lihongguang00 committed:
Merge branch 'lihongguang00-branch-Calendar-Model' into branch-Calendar-Model

lihongguang00 committed:
Resolve merge conflict

Resolved merge conflict

lihongguang00 committed:
Fixed checkstyle

Fixed checkstyle

junhonglow reviewed (approved):
LGTM
